### PR TITLE
updated script standards + added icons, default topic and surprise arceus battle

### PR DESCRIPTION
**Pull request by NightfallAlicorn**
all tested, any suggestions on improvements are welcome

### CODE REVIEW

File: scripts/amoebagame.js [hunk 1]
```diff
@@ -1,40 +1,70 @@
...
+                if (role < poke.length - 1) {
+                    SESSION.users(winner).amoeba.role++;
+                    if (SESSION.users(loser).amoeba.role > 0) {
+                        SESSION.users(loser).amoeba.role--;
+                    }
+                } else {
+                    SESSION.users(loser).amoeba.role = 0;
+                }
+                if (role === 0) {
+                    sys.sendHtmlAll("<hr><center>" + pokeIcon(role) + "<b>" + htmlEscape(sys.name(battles[role])) + "</b> and " + pokeIcon(role) + "<b>" + htmlEscape(sys.name(src)) + "</b> engaged in a fierce battle!</center><br /><b><font color='#0000ff'>" + htmlEscape(sys.name(winner)) + "</font></b> won and evolves into " + pokeIcon(SESSION.users(winner).amoeba.role) + poke[SESSION.users(winner).amoeba.role].species + " while <b><font color='#ff0000'>" + htmlEscape(sys.name(loser)) + "</font></b> lost and remains a " + pokeIcon(SESSION.users(loser).amoeba.role) + poke[SESSION.users(loser).amoeba.role].species + ".<hr>", id);
```
Comment (Fuzzysqurl) [inline scripts/amoebagame.js]: I feel like this just makes the channel more spammy....

File: scripts/amoebagame.js [hunk 2]
```diff
@@ -1,40 +1,70 @@
...
-                    + sys.name(loser) + " lost and is now a " + species[SESSION.users(loser).amoeba.role] + ".",
-                id);
+    this.beforeChatMessage = function (src, message, channel) {
+        if (channel !== id) {
+            return false;
+        }
+        var role = SESSION.users(src).amoeba.role,
+            isAuth = sys.auth(src) > 0 && sys.auth(src) < 4, // ignores hidden auth
+            name = htmlEscape(sys.name(src)),
+            messagePrefix = "<font color='" + script.getColor(src) + "'><timestamp/>" + pokeIcon(role) + (isAuth ? "+" : "") + "<b>" + (isAuth ? "<i>" + name + ":</i> " : name + ": ") + "</b></font>";
```
Comment (Fuzzysqurl) [inline scripts/amoebagame.js]: The icon seems redundant. I mean, you don't see it unless they talk. When they talk, they say what role they are. This just needlessly spaces out the chat.

File: scripts/amoebagame.js (hunk 1 above)
Comment (NightfallAlicorn) [inline scripts/amoebagame.js]: I was trying to keep it thin while adding more color and interest with the icons. If you know how to thin it down without using much effect that be great.

File: scripts/amoebagame.js (hunk 2 above)
Comment (NightfallAlicorn) [inline scripts/amoebagame.js]: I could shrink the icons a bit. Or, edit crop it out and change to base64 but it only make server send larger messages.

Comment (Fuzzysqurl) [inline scripts/amoebagame.js]: Shrinking would just make them look bad unfortunately. 

I'd honestly just remove this and leave your other change if we had to compromise. I'm against icons next to names cause then we look like some weird Spanish server :)

Comment (NightfallAlicorn) [inline scripts/amoebagame.js]: So remove all icon instances? Time stamps and battle?

File: scripts/amoebagame.js [hunk 3]
```diff
@@ -1,40 +1,70 @@
...
+                } else {
+                    sys.sendHtmlAll("<hr><center>" + /*pokeIcon(role) +*/ "<b>" + htmlEscape(sys.name(battles[role])) + "</b> and " + /*pokeIcon(role) +*/ "<b>" + htmlEscape(sys.name(src)) + "</b> engaged in a fierce battle of the Gods!</center><br /><b><font color='#0000ff'>" + htmlEscape(sys.name(winner)) + "</font></b> defeated <b><font color='#ff0000'>" + htmlEscape(sys.name(loser)) + "</font></b> and mercilessly reset their evolution back to " + /*pokeIcon(SESSION.users(loser).amoeba.role) +*/ poke[SESSION.users(loser).amoeba.role].species + ".<hr>", id);
```
Comment (Fuzzysqurl) [inline scripts/amoebagame.js]: This should also be removed. What's the point of getting to arceus if you get shot down too

Comment (NightfallAlicorn) [inline scripts/amoebagame.js]: Sulcuta suggested it. Thought it sounded like a good idea. :x So want me to remove Arceus battle too?

**Comment by NightfallAlicorn** [comment]
Moved to a single commit https://github.com/po-devs/po-server-goodies/pull/1315

**Pull request closed without merging**